jenkins/scripts/bare_metal_lab/tasks/cleanup-tasks.yaml

```diff
@@ -0,0 +1,128 @@
+- name: Check required env vars are set
+  fail:
+    msg: Ensure that BML_ILO_USERNAME, BML_ILO_PASSWORD and GITHUB_TOKEN environment variables are set
+  when: not bml_ilo_username or not bml_ilo_password or not github_token
+
+- name: Install dependencies for metal3-dev-env and hpilo_boot
+  apt:
+    pkg:
+      - make
+      - python3-hpilo
+  become: true
+
+- name: Add passwordless sudo
+  lineinfile:
+    path: "/etc/sudoers.d/{{ ansible_user_id}}_passwordless_sudo"
+    line: "{{ ansible_user_id }} ALL = (ALL) NOPASSWD: ALL"
+    create: true
+    owner: root
+    group: root
+    mode: 0400
+  become: true
+
+# Note: This file is persistent, so even though we do this *after* sending the
+# variables it is still helpful for the next run.
+- name: Ensure that iLO credentials and github token can be passed over SSH
+  lineinfile:
+    path: "/etc/ssh/sshd_config"
+    line: "AcceptEnv LANG LC_* BML_* GITHUB_TOKEN REPO_NAME"
+  become: true
+  register: sshd_config
+
+- name: Restart sshd if required
+  systemd:
+    name: sshd
+    state: restarted
+  become: true
+  when: sshd_config.changed
+
+- name: Shrink size of syslog
+  become: true
+  become_user: root
+  shell:
+    cmd: truncate --size 10737418240 /var/log/syslog && truncate --size 10737418240 /var/log/syslog.1   2>/dev/null
+  ignore_errors: true
+
+- name: Find old logs
+  ansible.builtin.find:
+    paths: /home/{{ ansible_user_id }}
+    recurse: no
+    patterns: "logs-jenkins-*"
+    file_type: any
+  register: old_logs
+  tags: deploy
+
+- name: Deleting files and folders from previous run
+  become: true
+  become_user: root
+  file:
+    state: absent
+    path: "{{ item }}"
+  with_items:
+    - "{{ old_logs.files | map(attribute='path') }}"
+    - /home/{{ ansible_user_id }}/go
+    - /home/{{ ansible_user_id }}/tested_repo
+    - /home/{{ ansible_user_id }}/metal3
+    - /home/{{ ansible_user_id }}/metal3-dev-env
+    - /opt/metal3-dev-env
+    - /tmp/mariadb.csr
+    - /tmp/ironic.csr
+    - /tmp/manifests
+    - "{{ serial_log_location }}"
+  tags: deploy
```

Comment:
if I understood right `--skip-tags "deploy"` in bml_cleanup.sh means we would skip the tasks with this tag, however we do need these to be run as part of the cleanup, are not we? also, `tags:deploy` inside the cleanup_tasks.yaml looks bit strange to me, how about renaming it to `tags:cleanup`?

Reply:
There are some cleanup tasks which we should not run for clean up after successful run. For example cleaning up logs, cloning repo, merging PR branch. These tasks should run when deploying new metal3 dev env.
If we move these tasks out of cleanup then tags would not necessarily needed, I will try to move these tasks out of cleanup script.